FormulaEvaluator broken on non UTF-8 systems

When loading the Javascript files from the class path to evaluate vertex titles, the encoding is not specified. On systems where the default Charset is not UTF-8, chrono.js fails to load.

We saw this problem with the tester on Guidewire. Able to reproduce locally by setting "-Dfile.encoding=Cp850" when running my web server. Produces the following stack trace:

Caused by: org.mozilla.javascript.EcmaError: SyntaxError: Invalid range in character class. (libs/chrono.min.js#1)
	at org.mozilla.javascript.ScriptRuntime.constructError(ScriptRuntime.java:3687)
	…
	at org.visallo.core.formula.FormulaEvaluator.evaluateFile(FormulaEvaluator.java:147)
	at org.visallo.core.formula.FormulaEvaluator.loadJavaScript(FormulaEvaluator.java:127)
	at org.visallo.core.formula.FormulaEvaluator.setupContext(FormulaEvaluator.java:117)
	at org.visallo.core.formula.FormulaEvaluator.getScriptable(FormulaEvaluator.java:89)
	at org.visallo.core.formula.FormulaEvaluator$FormulaEvaluatorCallable.call(FormulaEvaluator.java:199)
	at org.visallo.core.formula.FormulaEvaluator$FormulaEvaluatorCallable.call(FormulaEvaluator.java:184)
	…

Still there! 

2016-03-15 09:50:33.800/EDT ERROR [USER_4d49e3913fe54e3d8d9853d47f2f2cb7:nida] [core.formula.FormulaEvaluator] Error encountered during Title evaluation
java.util.concurrent.ExecutionException: org.mozilla.javascript.EcmaError: SyntaxError: Invalid range in character class. (libs/chrono.min.js#1)
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at org.visallo.core.formula.FormulaEvaluator.evaluateFormula(FormulaEvaluator.java:88)
	at org.visallo.core.formula.FormulaEvaluator.evaluateTitleFormula(FormulaEvaluator.java:63)
	at org.visallo.core.model.workspace.WorkspaceDiffHelper.createWorkspaceDiffVertexItem(WorkspaceDiffHelper.java:127)
	at org.visallo.core.model.workspace.WorkspaceDiffHelper.diffWorkspaceEntity(WorkspaceDiffHelper.java:117)
	at org.visallo.core.model.workspace.WorkspaceDiffHelper$$EnhancerByGuice$$94129c7.CGLIB$diffWorkspaceEntity$2(<generated>)
	at org.visallo.core.model.workspace.WorkspaceDiffHelper$$EnhancerByGuice$$94129c7$$FastClassByGuice$$cbaae114.invoke(<generated>)
	at com.google.inject.internal.cglib.proxy.$MethodProxy.invokeSuper(MethodProxy.java:228)
	at com.google.inject.internal.InterceptorStackCallback$InterceptedMethodInvocation.proceed(InterceptorStackCallback.java:72)
	at org.visallo.core.trace.TracedMethodInterceptor.invoke(TracedMethodInterceptor.java:20)
	at com.google.inject.internal.InterceptorStackCallback$InterceptedMethodInvocation.proceed(InterceptorStackCallback.java:72)
	at com.google.inject.internal.InterceptorStackCallback.intercept(InterceptorStackCallback.java:52)
	at org.visallo.core.model.workspace.WorkspaceDiffHelper$$EnhancerByGuice$$94129c7.diffWorkspaceEntity(<generated>)
	at org.visallo.core.model.workspace.WorkspaceDiffHelper.diff(WorkspaceDiffHelper.java:44)
	at org.visallo.core.model.workspace.WorkspaceDiffHelper$$EnhancerByGuice$$94129c7.CGLIB$diff$0(<generated>)
	at org.visallo.core.model.workspace.WorkspaceDiffHelper$$EnhancerByGuice$$94129c7$$FastClassByGuice$$cbaae114.invoke(<generated>)
	at com.google.inject.internal.cglib.proxy.$MethodProxy.invokeSuper(MethodProxy.java:228)
	at com.google.inject.internal.InterceptorStackCallback$InterceptedMethodInvocation.proceed(InterceptorStackCallback.java:72)
	at org.visallo.core.trace.TracedMethodInterceptor.invoke(TracedMethodInterceptor.java:20)
	at com.google.inject.internal.InterceptorStackCallback$InterceptedMethodInvocation.proceed(InterceptorStackCallback.java:72)
	at com.google.inject.internal.InterceptorStackCallback.intercept(InterceptorStackCallback.java:52)
	at org.visallo.core.model.workspace.WorkspaceDiffHelper$$EnhancerByGuice$$94129c7.diff(<generated>)
	at org.visallo.vertexium.model.workspace.VertexiumWorkspaceRepository$14.call(VertexiumWorkspaceRepository.java:772)
	at org.visallo.vertexium.model.workspace.VertexiumWorkspaceRepository$14.call(VertexiumWorkspaceRepository.java:765)
	at com.visallo.core.model.lock.CuratorLockRepository$1.run(CuratorLockRepository.java:45)
	at org.visallo.core.model.lock.LockRepository.lock(LockRepository.java:33)
	at org.visallo.vertexium.model.workspace.VertexiumWorkspaceRepository.getDiff(VertexiumWorkspaceRepository.java:765)
	at org.visallo.vertexium.model.workspace.VertexiumWorkspaceRepository$$EnhancerByGuice$$bed25d42.CGLIB$getDiff$3(<generated>)
	at org.visallo.vertexium.model.workspace.VertexiumWorkspaceRepository$$EnhancerByGuice$$bed25d42$$FastClassByGuice$$3473c205.invoke(<generated>)
	at com.google.inject.internal.cglib.proxy.$MethodProxy.invokeSuper(MethodProxy.java:228)
	at com.google.inject.internal.InterceptorStackCallback$InterceptedMethodInvocation.proceed(InterceptorStackCallback.java:72)
	at org.visallo.core.trace.TracedMethodInterceptor.invoke(TracedMethodInterceptor.java:20)
	at com.google.inject.internal.InterceptorStackCallback$InterceptedMethodInvocation.proceed(InterceptorStackCallback.java:72)
	at com.google.inject.internal.InterceptorStackCallback.intercept(InterceptorStackCallback.java:52)
	at org.visallo.vertexium.model.workspace.VertexiumWorkspaceRepository$$EnhancerByGuice$$bed25d42.getDiff(<generated>)
	at org.visallo.web.routes.workspace.WorkspaceDiff.handle(WorkspaceDiff.java:36)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.v5analytics.webster.RequestResponseHandlerParameterizedHandlerWrapper.handle(RequestResponseHandlerParameterizedHandlerWrapper.java:99)
	at com.v5analytics.webster.HandlerChain.next(HandlerChain.java:18)
	at org.visallo.web.privilegeFilters.PrivilegeFilter.handle(PrivilegeFilter.java:33)
	at com.v5analytics.webster.HandlerChain.next(HandlerChain.java:18)
	at com.v5analytics.webster.handlers.CSRFHandler.handle(CSRFHandler.java:66)
	at com.v5analytics.webster.HandlerChain.next(HandlerChain.java:18)
	at org.visallo.web.AuthenticationHandler.handle(AuthenticationHandler.java:17)
	at com.v5analytics.webster.HandlerChain.next(HandlerChain.java:18)
	at com.v5analytics.webster.Router.dispatch(Router.java:103)
	at com.v5analytics.webster.Router.routeWithExceptionHandling(Router.java:97)
	at com.v5analytics.webster.Router.route(Router.java:51)
	at com.v5analytics.webster.App.handle(App.java:133)
	at org.visallo.web.WebApp.handle(WebApp.java:109)
	at org.visallo.web.Router.service(Router.java:274)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:800)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1669)
	at org.eclipse.jetty.servlets.UserAgentFilter.doFilter(UserAgentFilter.java:83)
	at org.eclipse.jetty.servlets.GzipFilter.doFilter(GzipFilter.java:364)
	at org.visallo.web.ApplicationGzipFilter.doFilter(ApplicationGzipFilter.java:20)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at org.visallo.web.RequestDebugFilter.doFilter(RequestDebugFilter.java:34)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:585)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:577)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:215)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
	at org.eclipse.jetty.server.Server.handle(Server.java:497)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:310)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:245)
	at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.mozilla.javascript.EcmaError: SyntaxError: Invalid range in character class. (libs/chrono.min.js#1)
	at org.mozilla.javascript.ScriptRuntime.constructError(ScriptRuntime.java:3687)
	at org.mozilla.javascript.ScriptRuntime.constructError(ScriptRuntime.java:3665)
	at org.mozilla.javascript.regexp.NativeRegExp.reportError(NativeRegExp.java:2540)
	at org.mozilla.javascript.regexp.NativeRegExp.calculateBitmapSize(NativeRegExp.java:669)
	at org.mozilla.javascript.regexp.NativeRegExp.parseTerm(NativeRegExp.java:1055)
	at org.mozilla.javascript.regexp.NativeRegExp.parseAlternative(NativeRegExp.java:521)
	at org.mozilla.javascript.regexp.NativeRegExp.parseDisjunction(NativeRegExp.java:452)
	at org.mozilla.javascript.regexp.NativeRegExp.parseTerm(NativeRegExp.java:1014)
	at org.mozilla.javascript.regexp.NativeRegExp.parseAlternative(NativeRegExp.java:521)
	at org.mozilla.javascript.regexp.NativeRegExp.parseDisjunction(NativeRegExp.java:452)
	at org.mozilla.javascript.regexp.NativeRegExp.parseTerm(NativeRegExp.java:1014)
	at org.mozilla.javascript.regexp.NativeRegExp.parseAlternative(NativeRegExp.java:521)
	at org.mozilla.javascript.regexp.NativeRegExp.parseDisjunction(NativeRegExp.java:452)
	at org.mozilla.javascript.regexp.NativeRegExp.parseDisjunction(NativeRegExp.java:461)
	at org.mozilla.javascript.regexp.NativeRegExp.parseTerm(NativeRegExp.java:1014)
	at org.mozilla.javascript.regexp.NativeRegExp.parseAlternative(NativeRegExp.java:521)
	at org.mozilla.javascript.regexp.NativeRegExp.parseDisjunction(NativeRegExp.java:452)
	at org.mozilla.javascript.regexp.NativeRegExp.parseTerm(NativeRegExp.java:1014)
	at org.mozilla.javascript.regexp.NativeRegExp.parseAlternative(NativeRegExp.java:521)
	at org.mozilla.javascript.regexp.NativeRegExp.parseDisjunction(NativeRegExp.java:452)
	at org.mozilla.javascript.regexp.NativeRegExp.compileRE(NativeRegExp.java:311)
	at org.mozilla.javascript.regexp.RegExpImpl.compileRegExp(RegExpImpl.java:22)
	at org.mozilla.javascript.gen.libs_chrono_min_js_12._reInit(libs/chrono.min.js)
	at org.mozilla.javascript.gen.libs_chrono_min_js_12._i1(libs/chrono.min.js)
	at org.mozilla.javascript.gen.libs_chrono_min_js_12.<init>(libs/chrono.min.js)
	at org.mozilla.javascript.gen.libs_chrono_min_js_12._c_script_0(libs/chrono.min.js:1)
	at org.mozilla.javascript.gen.libs_chrono_min_js_12.call(libs/chrono.min.js)
	at org.mozilla.javascript.gen.libs_chrono_min_js_12.exec(libs/chrono.min.js)
	at org.mozilla.javascript.Context.evaluateString(Context.java:1079)
	at org.visallo.core.formula.RequireJsSupport.processSource(RequireJsSupport.java:64)
	at org.visallo.core.formula.RequireJsSupport.load(RequireJsSupport.java:48)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.mozilla.javascript.MemberBox.invoke(MemberBox.java:126)
	at org.mozilla.javascript.FunctionObject.call(FunctionObject.java:386)
	at org.mozilla.javascript.optimizer.OptRuntime.callName(OptRuntime.java:63)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_133(libs/r.js:2367)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callN(OptRuntime.java:52)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_111(libs/r.js:1885)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.call2(OptRuntime.java:42)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_67(libs/r.js:1066)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callProp0(OptRuntime.java:85)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_65(libs/r.js:1056)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callProp0(OptRuntime.java:85)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_68(libs/r.js:1086)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callProp0(OptRuntime.java:85)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_80(libs/r.js:1389)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callProp0(OptRuntime.java:85)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_108(libs/r.js:1757)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.call2(OptRuntime.java:42)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_81(libs/r.js:1374)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.ScriptRuntime.applyOrCall(ScriptRuntime.java:2430)
	at org.mozilla.javascript.BaseFunction.execIdCall(BaseFunction.java:269)
	at org.mozilla.javascript.IdFunctionObject.call(IdFunctionObject.java:97)
	at org.mozilla.javascript.optimizer.OptRuntime.call2(OptRuntime.java:42)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_30(libs/r.js:370)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callN(OptRuntime.java:52)
	at org.mozilla.javascript.gen.libs_r_js_6._c_each_22(libs/r.js:295)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callName(OptRuntime.java:63)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_80(libs/r.js:1336)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callProp0(OptRuntime.java:85)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_62(libs/r.js:1020)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.call2(OptRuntime.java:42)
	at org.mozilla.javascript.gen.libs_r_js_6._c_callGetModule_87(libs/r.js:1416)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callName(OptRuntime.java:63)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_109(libs/r.js:1790)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.call1(OptRuntime.java:32)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_133(libs/r.js:2370)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callN(OptRuntime.java:52)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_111(libs/r.js:1885)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.call2(OptRuntime.java:42)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_67(libs/r.js:1066)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callProp0(OptRuntime.java:85)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_65(libs/r.js:1056)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callProp0(OptRuntime.java:85)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_68(libs/r.js:1086)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callProp0(OptRuntime.java:85)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_80(libs/r.js:1389)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callProp0(OptRuntime.java:85)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_108(libs/r.js:1757)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.call2(OptRuntime.java:42)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_81(libs/r.js:1374)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.ScriptRuntime.applyOrCall(ScriptRuntime.java:2430)
	at org.mozilla.javascript.BaseFunction.execIdCall(BaseFunction.java:269)
	at org.mozilla.javascript.IdFunctionObject.call(IdFunctionObject.java:97)
	at org.mozilla.javascript.optimizer.OptRuntime.call2(OptRuntime.java:42)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_30(libs/r.js:370)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callN(OptRuntime.java:52)
	at org.mozilla.javascript.gen.libs_r_js_6._c_each_22(libs/r.js:295)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callName(OptRuntime.java:63)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_80(libs/r.js:1336)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callProp0(OptRuntime.java:85)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_62(libs/r.js:1020)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.call2(OptRuntime.java:42)
	at org.mozilla.javascript.gen.libs_r_js_6._c_callGetModule_87(libs/r.js:1416)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callName(OptRuntime.java:63)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_109(libs/r.js:1790)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.call1(OptRuntime.java:32)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_133(libs/r.js:2370)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callN(OptRuntime.java:52)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_111(libs/r.js:1885)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.call2(OptRuntime.java:42)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_67(libs/r.js:1066)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callProp0(OptRuntime.java:85)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_65(libs/r.js:1056)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callProp0(OptRuntime.java:85)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_68(libs/r.js:1086)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callProp0(OptRuntime.java:85)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_80(libs/r.js:1389)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callProp0(OptRuntime.java:85)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_108(libs/r.js:1757)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.call2(OptRuntime.java:42)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_81(libs/r.js:1374)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.ScriptRuntime.applyOrCall(ScriptRuntime.java:2430)
	at org.mozilla.javascript.BaseFunction.execIdCall(BaseFunction.java:269)
	at org.mozilla.javascript.IdFunctionObject.call(IdFunctionObject.java:97)
	at org.mozilla.javascript.optimizer.OptRuntime.call2(OptRuntime.java:42)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_30(libs/r.js:370)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callN(OptRuntime.java:52)
	at org.mozilla.javascript.gen.libs_r_js_6._c_each_22(libs/r.js:295)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callName(OptRuntime.java:63)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_80(libs/r.js:1336)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callProp0(OptRuntime.java:85)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_62(libs/r.js:1020)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callN(OptRuntime.java:52)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_102(libs/r.js:1662)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.call0(OptRuntime.java:23)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_118(libs/r.js:2002)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.call1(OptRuntime.java:32)
	at org.mozilla.javascript.gen.libs_r_js_6._c_localRequire_101(libs/r.js:1651)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callN(OptRuntime.java:52)
	at org.mozilla.javascript.gen.libs_r_js_6._c_anonymous_115(libs/r.js:1983)
	at org.mozilla.javascript.gen.libs_r_js_6.call(libs/r.js)
	at org.mozilla.javascript.optimizer.OptRuntime.callName(OptRuntime.java:63)
	at org.mozilla.javascript.gen.loader_js_8._c_script_0(loader.js:57)
	at org.mozilla.javascript.gen.loader_js_8.call(loader.js)
	at org.mozilla.javascript.ContextFactory.doTopCall(ContextFactory.java:394)
	at org.mozilla.javascript.ScriptRuntime.doTopCall(ScriptRuntime.java:3091)
	at org.mozilla.javascript.gen.loader_js_8.call(loader.js)
	at org.mozilla.javascript.gen.loader_js_8.exec(loader.js)
	at org.mozilla.javascript.Context.evaluateString(Context.java:1079)
	at org.visallo.core.formula.FormulaEvaluator.evaluateFile(FormulaEvaluator.java:162)
	at org.visallo.core.formula.FormulaEvaluator.loadJavaScript(FormulaEvaluator.java:142)
	at org.visallo.core.formula.FormulaEvaluator.setupContext(FormulaEvaluator.java:132)
	at org.visallo.core.formula.FormulaEvaluator.getScriptable(FormulaEvaluator.java:104)
	at org.visallo.core.formula.FormulaEvaluator$FormulaEvaluatorCallable.call(FormulaEvaluator.java:218)
	at org.visallo.core.formula.FormulaEvaluator$FormulaEvaluatorCallable.call(FormulaEvaluator.java:199)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	... 1 more

--- a/core/core/src/main/java/org/visallo/core/formula/RequireJsSupport.java
+++ b/core/core/src/main/java/org/visallo/core/formula/RequireJsSupport.java
@@ -11,6 +11,7 @@ import org.mozilla.javascript.ScriptableObject;
 import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.io.InputStream;
+import java.nio.charset.Charset;
 
 public class RequireJsSupport extends ScriptableObject {
 
@@ -68,7 +69,7 @@ public class RequireJsSupport extends ScriptableObject {
         InputStream is = RequireJsSupport.class.getResourceAsStream(file);
         if (is != null) {
             try {
-                return IOUtils.toString(is);
+                return IOUtils.toString(is, Charset.forName("UTF-8"));
             } catch (IOException e) {
                 LOGGER.error("File not readable %s", file, e);
             }

